### DISCUSSION TITLE
GMCP Requests (Aardwolf)

### DISCUSSION THREAD

**Discussion by thechanster** [scandum/tintin#192] [Script help]
Hi Scandum and all,

Thanks as always for your prompt help.
I have GMCP working fine for Aardwolf when server sends info, but looking to "request" info from the server.
There is very little documentation on sending a request vs receiving a request. Have the usual codes created, and as far as I can tell, something like the following is on the right track.

```
#format IAC  %a 255
#format DONT %a 254
#format DO   %a 253
#format WONT %a 252
#format WILL %a 251
#format SB   %a 250
#format SE   %a 240
#format GMCP %a 201

#ALIAS {gmcp %0} {#send {${IAC}${SB}${GMCP} %0 ${IAC}${SE}\}}
```
I'm really unclear what the %0 format should be, if spaces before and after are bad, etc., and I think Aardwolf request structure is confusing as it seems to be two words separated by a space, not dot notation.

Command:
request area
request char
request quest
request room
request sectors

Can anyone shed some light on how I might get this working?

Also not sure how to get debug info on a failed command, is there a way to get that info?

Regards, Matt

**Comment by scandum** [comment]
You may have to drop the spaces before/after.

You can get additional info using `#config debug_telnet on`, might help to catch some unusual responses.

**Comment by thechanster** [comment]
Hey, thanks again, I didn't know about telnet debug. 

I have managed to get it working, somewhere in my code (someone elses code i was using) broke the request, so after first request, all subsequent ones failed. After removing the offending code (haven't reviewed it yet) requests are working again on demand.
